### PR TITLE
Save and load data from CoreData

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Fix - save data to CoreData and update UI]
## WHAT
- check connection
- Save weather data to coredata
- Load data local when not connected
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: Weather_App/Weather_App/Scenes/Map/MapViewController.swift [hunk 1]
```diff
@@ -118,7 +113,9 @@ extension MapViewController {
             case .success(let weatherCurrent):
                 DispatchQueue.main.async { [weak self] in
                     guard let self = self else { return }
-                    self.updateUIWithData(weatherCurrent)
+                    self.updateUIWithAPIData(weatherCurrent)
+                    weatherCurrentCoreDataManager.saveWeatherToCoreData(weatherCurrent: weatherCurrent)
+                    nameCitySaved = weatherCurrent.nameCity
```
Comment (ghost) [inline Weather_App/Weather_App/Scenes/Map/MapViewController.swift]: thêm self e

File: Weather_App/Weather_App/Scenes/Map/MapViewController.swift [hunk 2]
```diff
@@ -164,8 +142,49 @@ extension MapViewController {
...
+            configureMap()
+        } else {
+            self.presentErrorAlert(title: "ERROR", message: "Not Connected")
+            if let savedWeatherData = weatherCurrentCoreDataManager.fetchWeatherEntity() {
+                for weatherEntity in savedWeatherData {
+                    if weatherEntity.userLocation == true {
+                        updateUIWithCoreData(weatherEntity)
+                        break
+                    }
+                }
```
Comment (ghost) [inline Weather_App/Weather_App/Scenes/Map/MapViewController.swift]: dùng for ... where e

File: Weather_App/Weather_App/Services/Local/WeatherCurrentCoreDataManager.swift [hunk 3]
```diff
@@ -26,24 +38,93 @@ class WeatherCurrentCoreDataManager {
...
+        let fetchRequest: NSFetchRequest<WeatherEntity> = WeatherEntity.fetchRequest()
+        fetchRequest.predicate = NSPredicate(format: "nameCity == %@ AND userLocation == true", cityName)
+        
+        do {
+            let results = try context.fetch(fetchRequest)
+            for (index, entity) in results.enumerated() {
+                if index > 0 {
+                    context.delete(entity)
+                }
+            }
```
Comment (ghost) [inline Weather_App/Weather_App/Services/Local/WeatherCurrentCoreDataManager.swift]: for where lun e

File: Weather_App/Weather_App/Services/Remote/NetworkMonitor.swift [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+import Network
+
+final class NetworkMonitor {
+    static let shared = NetworkMonitor()
+    
+    private let queue = DispatchQueue.global()
+    private let monitor = NWPathMonitor()
+    
+    private var isConnected: Bool = false
```
Comment (ghost) [inline Weather_App/Weather_App/Services/Remote/NetworkMonitor.swift]:
```suggestion
    private var isConnected = false
```

**Review by ghost — APPROVED** [review]

**Review by daolq-2712 — APPROVED** [review]
OK!

**Pull request merged**